src/items/LinkItem.tsx

```diff
@@ -92,98 +98,108 @@ const LinkItem: FC<LinkItemProps> = ({
     edit: editCaption,
   });
 
-  // if available, display specific player
-  const html = extra?.html;
-  if (html) {
-    const component = (
-      <div id={id} dangerouslySetInnerHTML={{ __html: html }} />
-    );
-    if (showCaption) {
-      return CaptionWrapper(component);
+  const getComponent = (): JSX.Element | null => {
+    // if available, display specific player
+    const html = extra?.html;
+    if (html) {
+      const component = (
+        <div id={id} dangerouslySetInnerHTML={{ __html: html }} />
+      );
+      if (showCaption) {
+        return CaptionWrapper(component);
+      }
```

Comment:
I think we want to show the caption each time so it makes sense to apply it outside of the function, before show collapse.